### PR TITLE
MODINVOSTO-26 Implement CRUD for /invoice-storage/acquisitions-unit-assignments

### PR DESCRIPTION
**Pull request by grabsefx**
## Purpose
https://issues.folio.org/browse/MODINVOSTO-26

## Approach
* ModuleDescriptor, RAML are updated
* Implemented CRUD for /invoice-storage/acquisitions-unit-assignments
* Index added on recordId field
* Unit tests refactored to use junit5 features
* Eclipse style formatted code

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/rest/impl/TenantReferenceAPI.java [hunk 1]
```diff
@@ -64,7 +67,8 @@ private boolean buildDataLoadingParameters(TenantAttributes tenantAttributes, Te
         .add("invoices", "invoice-storage/invoices")
         .add("invoice-lines", "invoice-storage/invoice-lines")
         .add("vouchers", "voucher-storage/vouchers")
-        .add("voucher-lines", "voucher-storage/voucher-lines");
+        .add("voucher-lines", "voucher-storage/voucher-lines")
+        .add("acquisitions-unit-assignments", "orders-storage/acquisitions-unit-assignments");
```
Comment (piotr-kalashuk) [inline src/main/java/org/folio/rest/impl/TenantReferenceAPI.java]: I believe `orders-storage` should be renamed to `invoice-storage`

Comment (grabsefx) [reply src/main/java/org/folio/rest/impl/TenantReferenceAPI.java]: fixed

File: src/test/java/org/folio/rest/impl/EntitiesCrudTest.java [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+      TestEntities.VOUCHER_LINES,
+      TestEntities.VOUCHER,
+      TestEntities.INVOICE_LINES);
+  }
+
+  @ParameterizedTest
+  @Order(1)
+  @EnumSource(TestEntities.class)
+  void testVerifyCollection(TestEntities testEntity) throws MalformedURLException {
+    logger.info(String.format("--- mod-orders-storage %s test: Verifying database's initial state ... ", testEntity.name()));
```
Comment (piotr-kalashuk) [inline src/test/java/org/folio/rest/impl/EntitiesCrudTest.java]: Please remove or replace `mod-orders-storage` by `mod-invoice-storage`. This is applicable for all log messages in this class

Comment (grabsefx) [reply src/test/java/org/folio/rest/impl/EntitiesCrudTest.java]: fixed

File: ramls/acquisitions-unit-assignments.raml [hunk 3]
```diff
@@ -0,0 +1,56 @@
+#%RAML 1.0
+title: "Invoice Storage"
+baseUri: http://github.com/folio-org/mod-invoice-storage
+version: v1
+
+documentation:
+  - title: Acquisitions units
```
Comment (KVupp) [inline ramls/acquisitions-unit-assignments.raml]: please rename title to acquisitions units assignments

File: pom.xml [hunk 4]
```diff
@@ -117,6 +136,23 @@
...
+            <artifactId>junit-platform-surefire-provider</artifactId>
+            <version>1.2.0</version>
+          </dependency>
+          <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-engine</artifactId>
+            <version>${junit-jupiter-version}</version>
+          </dependency>
+          <dependency>
+            <groupId>org.junit.vintage</groupId>
```
Comment (KVupp) [inline pom.xml]: Optional : As all the unit tests are moved to junit 5 , I don't think vintage jar is needed

Comment (piotr-kalashuk) [reply pom.xml]: Also `<groupId>junit</groupId>` dependency can be removed

**Pull request merged**